main.tf

```diff
@@ -24,153 +21,188 @@ resource "aws_eks_cluster" "main" {
     }
   }
 
-
   dynamic "encryption_config" {
     for_each = var.encryption_config
     content {
       dynamic "provider" {
         for_each = lookup(encryption_config.value, "provider")
         content {
-          key_arn = provider.value.key_arn
+          key_arn = lookup(provider.value.key_arn, module.secrets_kms_key[0].arn)
         }
       }
-      resources = encryption_config.value.resources
+      resources = ["secrets"]
     }
   }
-}
-
-/*
-eks addon
-*/
 
-resource "aws_eks_addon" "main" {
-  count                    = var.create_eks_addon ? 1 : 0
-  cluster_name             = aws_eks_cluster.main.name
-  addon_name               = var.addon_name
-  addon_version            = var.addon_version
-  resolve_conflicts        = var.resolve_conflicts
-  tags                     = var.tags
-  service_account_role_arn = var.service_account_role_arn
+  # Ensure that IAM Role permissions are created before and deleted after EKS Cluster handling.
+  # Otherwise, EKS will not be able to properly delete EKS managed EC2 infrastructure such as Security Groups.
+  depends_on = [
+    aws_iam_role_policy_attachment.amazoneksclusterpolicy,
+    aws_iam_role_policy_attachment.amazoneksvpccontroller,
+    aws_cloudwatch_log_group.main
+  ]
 }
 
-/*
-eks node group
-*/
-
-resource "aws_eks_node_group" "main" {
-  count                  = var.create_eks_node_group ? 1 : 0
-  cluster_name           = aws_eks_cluster.main.name
-  node_role_arn          = var.node_role_arn
-  subnet_ids             = var.node_group_subnet_ids
-  ami_type               = var.ami_type
-  capacity_type          = var.capacity_type
-  disk_size              = var.disk_size
-  force_update_version   = var.force_update_version
-  instance_types         = var.instance_types
-  labels                 = var.labels
-  node_group_name        = var.node_group_name_prefix != null ? null : var.node_group_name
-  node_group_name_prefix = var.node_group_name != null ? null : var.node_group_name_prefix
-  release_version        = var.release_version
-  tags                   = var.tags
-  version                = var.kubernetes_version
-
-  dynamic "scaling_config" {
-    for_each = var.scaling_config
-    content {
-      desired_size = scaling_config.value.desired_size
-      max_size     = scaling_config.value.max_size
-      min_size     = scaling_config.value.min_size
-    }
-  }
+module "secrets_kms_key" {
+  count               = var.encryption_config == null ? 1 : 0
```

Comment:
A module calling an external module creates external dependencies we don't want to manage, also , since there isn't a version here, this would mean that we would pull the latest version, if major changes where done to the kmns module there is the high probability we would break this module.

This is a no no - please add the kms key within the module, and, allow to use a standard key or a module key or a external key.